bot/cogs/snekbox.py

```diff
@@ -199,32 +203,68 @@ def get_status_emoji(results: dict) -> str:
 
         log.info(f"Received code from {ctx.author} for evaluation:\n{code}")
 
-        self.jobs[ctx.author.id] = datetime.datetime.now()
-        code = self.prepare_input(code)
+        _predicate_eval_message_edit = partial(predicate_eval_message_edit, ctx)
+        _predicate_emoji_reaction = partial(predicate_eval_emoji_reaction, ctx)
 
-        try:
-            async with ctx.typing():
-                results = await self.post_eval(code)
-                msg, error = self.get_results_message(results)
-
-                if error:
-                    output, paste_link = error, None
-                else:
-                    output, paste_link = await self.format_output(results["stdout"])
-
-                icon = self.get_status_emoji(results)
-                msg = f"{ctx.author.mention} {icon} {msg}.\n\n```py\n{output}\n```"
-                if paste_link:
-                    msg = f"{msg}\nFull output: {paste_link}"
-
-                response = await ctx.send(msg)
-                self.bot.loop.create_task(
-                    wait_for_deletion(response, user_ids=(ctx.author.id,), client=ctx.bot)
+        while True:
+            self.jobs[ctx.author.id] = datetime.datetime.now()
+            code = self.prepare_input(code)
+
+            try:
+                async with ctx.typing():
+                    results = await self.post_eval(code)
+                    msg, error = self.get_results_message(results)
+
+                    if error:
+                        output, paste_link = error, None
+                    else:
+                        output, paste_link = await self.format_output(results["stdout"])
+
+                    icon = self.get_status_emoji(results)
+                    msg = f"{ctx.author.mention} {icon} {msg}.\n\n```py\n{output}\n```"
+                    if paste_link:
+                        msg = f"{msg}\nFull output: {paste_link}"
+
+                    response = await ctx.send(msg)
+                    self.bot.loop.create_task(
+                        wait_for_deletion(response, user_ids=(ctx.author.id,), client=ctx.bot)
+                    )
+
+                    log.info(f"{ctx.author}'s job had a return code of {results['returncode']}")
+            finally:
+                del self.jobs[ctx.author.id]
+
+            try:
+                _, new_message = await self.bot.wait_for(
+                    'message_edit',
+                    check=_predicate_eval_message_edit,
+                    timeout=10
                 )
+                await ctx.message.add_reaction('🔁')
```

Comment:
Using a unicode literal like this can sometimes be problematic. What you see in other parts of the bot is that these emojis are defined as constants that use unicode code points instead (like `"\U0001F195"`), because this tends to be a safer way of dealing with them. Some users have reported running into weird errors when the actual emoji is left inside of code like this.